src/main/java/com/carrotcreative/recyclercore/adapter/RecyclerCoreAdapter.java

```diff
@@ -1,77 +1,54 @@
 package com.carrotcreative.recyclercore.adapter;
 
 import android.support.v7.widget.RecyclerView;
+import android.util.SparseArray;
 import android.view.LayoutInflater;
+import android.view.View;
 import android.view.ViewGroup;
 
+import com.carrotcreative.recyclercore.inject.InjectController;
 import com.carrotcreative.recyclercore.util.InstantiationUtil;
+
 import java.util.HashMap;
 import java.util.List;
 
-public class RecyclerCoreAdapter extends RecyclerView.Adapter<RecyclerCoreController>{
+public class RecyclerCoreAdapter<ModelType> extends RecyclerView.Adapter<RecyclerCoreController>
```

Comment:
I don't suspect any issues arising for using Object here.

But I also don't suspect any issues occurring with the way that you've got it set up currently... I'll let you make the call!